test/verify/check-machines

```diff
@@ -370,6 +380,71 @@ class TestMachines(MachineCase):
             print("Libvirt version does not support disk statistics")
             b.wait_present("#vm-{0}-disksstats-unsupported".format(name))
 
+    def testLibvirt(self):
+        b = self.browser
+        m = self.machine
+
+        libvirtServiceName = self.getLibvirtName()
+
+        def checkLibvirtEnabled():
+            try:
+                m.execute("systemctl -q is-enabled {0}".format(libvirtServiceName))
+                return True
+            except:  # exception if return code != 0
+                return False
+
+        self.startVm("subVmTest1")
+        self.login_and_go("/machines")
+
+        b.wait_in_text("body", "Virtual Machines")
+        b.wait_present("tbody tr th")
+        b.wait_in_text("tbody tr th", "subVmTest1")
+
+        b.wait(lambda: checkLibvirtEnabled())
+        m.execute("systemctl disable {0}".format(libvirtServiceName))
+        print(self.getLibvirtName())
+        m.execute("systemctl stop {0}".format(libvirtServiceName))
+        b.wait(lambda: not checkLibvirtEnabled())
+        b.wait_present("#slate-header")
+        b.wait_in_text("#slate-header", "Virtualization Service (libvirt) is Not Active")
+        b.wait_present("#enable-libvirt:not(:checked)")
+        b.set_checked("#enable-libvirt", True)
+        b.click("#start-libvirt")
+
+        b.wait_in_text("body", "Virtual Machines")
+        b.wait(lambda: checkLibvirtEnabled())
+        b.wait_present("tbody tr th")
+        b.wait_in_text("tbody tr th", "subVmTest1")
+
+        m.execute("systemctl stop {0}".format(libvirtServiceName))
+        b.wait_present("#slate-header")
+        b.wait_in_text("#slate-header", "Virtualization Service (libvirt) is Not Active")
+        b.wait_present("#enable-libvirt:checked")
+        b.set_checked("#enable-libvirt", False)
+        b.click("#start-libvirt")
+        b.wait(lambda: not checkLibvirtEnabled())
+        b.wait_in_text("body", "Virtual Machines")
+        b.wait_present("tbody tr th")
+        b.wait_in_text("tbody tr th", "subVmTest1")
+
+        m.execute("systemctl enable {0}".format(libvirtServiceName))
+        m.execute("systemctl stop {0}".format(libvirtServiceName))
+        b.wait(lambda: checkLibvirtEnabled())
+        b.wait_present("#slate-header")
+        b.wait_in_text("#slate-header", "Virtualization Service (libvirt) is Not Active")
+        b.wait_present("#enable-libvirt:checked")
+
+        b.click("#troubleshoot")
+        b.leave_page()
+        url_location = "/system/services#/{0}".format(libvirtServiceName)
+        b.wait(lambda: url_location in b.eval_js("window.location.href"))
+
+        # HACK: restarting libvirtd causes this completely unrelated SELinux issue
+        self.allow_journal_messages('.*denied.*search.*"systemd-machine".*dev="proc".*')
```

Comment:
We generally report them to bugzilla, create a [known issue](https://github.com/cockpit-project/cockpit/issues?q=is%3Aissue+is%3Aopen+label%3Aknownissue), and add a [naughty override](https://github.com/cockpit-project/cockpit/tree/master/bots/naughty) for these, so that these issues get fixed and these hacks clean up themselves over time.

Reply:
I am pretty sure it was complaining about creating/removing symlinks, but now it is fine on my setup. 

It only throws this on ubuntu-1604
``` 
audit: type=1400 audit(1521636424.736:15): apparmor="DENIED" operation="open" profile="libvirt-a73c87e3-37a7-43c0-8671-b477f5783d95" name="/etc/gss/mech.d/" pid=2069 comm="qemu-system-x86" requested_mask="r" denied_mask="r" fsuid=64055 ouid=0
```